lib/datalog/roletester/src/role_tester.rs

```diff
@@ -107,8 +106,7 @@ fn create_error_ptr(e: String) -> Output {
 #[no_mangle]
 pub unsafe extern "C" fn process_access(input: *mut c_uchar, input_len: size_t) -> *mut Output {
     let mut runtime = Crepe::new();
-    let b = slice::from_raw_parts_mut(input, input_len);
-    let r = match types::Facts::decode(BytesMut::from(&b[..])) {
+    let r = match types::Facts::decode(&*slice::from_raw_parts_mut(input, input_len)) {
```

Comment:
@xacrimon @espadolini does this look correct to you?

Reply:
Can we make an immutable slice here instead of a mutable one?

Reply:
```suggestion
    let r = match types::Facts::decode(slice::from_raw_parts(input, input_len)) {
```